core/new-gui/src/app/workspace/service/cache-workflow/cache-workflow.service.ts

```diff
@@ -115,55 +111,72 @@ export class CacheWorkflowService {
       this.workflowActionService.getTexeraGraph().getBreakpointChangeStream(),
       this.workflowActionService.getJointGraphWrapper().getOperatorPositionChangeEvent()
     ).debounceTime(100).subscribe(() => {
-      const workflow = this.workflowActionService.getTexeraGraph();
+      const workflow1 = this.workflowActionService.getTexeraGraph();
 
-      const operators = workflow.getAllOperators();
-      const links = workflow.getAllLinks();
+      const operators = workflow1.getAllOperators();
+      const links = workflow1.getAllLinks();
       const operatorPositions: { [key: string]: Point } = {};
-      const breakpointsMap = workflow.getAllLinkBreakpoints();
+      const breakpointsMap = workflow1.getAllLinkBreakpoints();
       const breakpoints: Record<string, Breakpoint> = {};
       breakpointsMap.forEach((value, key) => (breakpoints[key] = value));
-      workflow.getAllOperators().forEach(op => operatorPositions[op.operatorID] =
+      workflow1.getAllOperators().forEach(op => operatorPositions[op.operatorID] =
         this.workflowActionService.getJointGraphWrapper().getOperatorPosition(op.operatorID));
 
-      const cachedWorkflow: CachedWorkflow = {
+      const cachedWorkflow: WorkflowInfo = {
         operators, operatorPositions, links, breakpoints
       };
-
-      this.setCachedWorkflow(JSON.stringify(cachedWorkflow));
+      let workflow: Workflow | null = this.getCachedWorkflow();
+      if (workflow == null) {
+        workflow = CacheWorkflowService.DEFAULT_WORKFLOW;
+      }
+      workflow.content = cachedWorkflow;
+      this.cacheWorkflow(workflow);
     });
   }
 
-  public getCachedWorkflow(): string | null {
-    return localStorage.getItem(CacheWorkflowService.LOCAL_STORAGE_KEY);
+  public getCachedWorkflow(): Workflow | null {
```

Comment:
wrap `null` to be `undefined`, since we use `undefined` pretty much everywhere else